socorro/signature/siglists/irrelevant_signature_re.txt

```diff
@@ -116,11 +116,19 @@ mozilla::gfx::Log<
 mozilla::ipc::IPDLParamTraits<T>::Write
 mozilla::ipc::WriteIPDLParam
 mozilla::Monitor::Wait
+mozilla::MozPromiseRefcountable::Release
+mozilla::MozPromiseRequestHolder<T>::Track
+mozilla::MozPromise<T>::AssertIsDead
+mozilla::MozPromise<T>::~MozPromise
+mozilla::MozPromise<T>::Private::~Private
+mozilla::MozPromise<T>::ThenCommand<T>::Track
+mozilla::MozPromise<T>::ThenInternal
+mozilla::MozPromise<T>::ThenValueBase::AssertIsDead
 mozilla::net::ShutdownEvent::PostAndWait
 mozilla::OffTheBooksCondVar::Wait
 mozilla::RefPtrTraits<T>::
-mozilla::TaskQueue::
 mozilla::TaskController::GetRunnableForMTTask
+mozilla::TaskQueue::
```

Comment:
I keep thinking about writing a sorter for these files so we have something that does it automatically and checks that the files are sorted in CI. Do you think we're at the point where that'd be helpful to have?

Reply:
Yeah, we should think about that. Some of the stuff in there is grouped by "theme" but as time goes by and the signatures evolve these group often lose their meanings... and having the files sorted means that adding stuff is easier.

Reply:
We could have a "section delimiter" so it's only sorting within the section.

What we really want is a way to tie documentation to groups of frames. I'm pretty sure the ordering in the file doesn't affect what happens in signature generation. Maybe instead of having all the irrelevant things in one file, we do a `.d` directory with a bunch of files in it? Then each file can be a sorted logical grouping of strings that are irrelevant.

It might be nice to be able to support other kinds of matching, too. Maybe "match this if this other thing about the crash report is true"? Or maybe "match this if the type starts with xyz"? I'm pretty sure that's come up before, but I never kept track of specific examples.

I'll write up a bug.